### PR TITLE
Aligned START expression with OSPool glideins. Added support for DESI…

### PR DESCRIPTION
**Pull request by rynge**
…RED_Sites/UNDESIRED_Sites (SOFTWARE-4874), IsBlackHole (replaces old RecentJobStarts approach), and HasExcessiveLoad.

### CODE REVIEW

**Comment by dsschult** [comment]
Is there a reason for why the DESIRED_Sites and UNDESIRED_Sites are being matched on the startd instead of in the job requirements?  I'm curious if there are good reasons for this, from either policy or HTCondor performance perspectives (I thought it was more performant to put these in the job requirements).

**Comment by rynge** [comment]
For the glideins, DESIRED_Sites/UNDESIRED_Sites are also used for signaling what sites to provision the glidines for. It is basically a way to target just a subset of sites within a group. The problem for the OSPool is that for DESIRED_Sites/UNDESIRED_Sites to work correctly, all startds have to use them. That is why we need them here on the containers as well.

File: 50-main.config [hunk 1]
```diff
@@ -41,14 +41,22 @@ GLIDEIN_ResourceName = "Docker"
...
+               IsBlackHole HasExcessiveLoad
+
+IsBlackHole = IfThenElse(RecentJobDurationAvg is undefined, false, RecentJobDurationCount >= 10 && RecentJobDurationAvg < 180)
+HasExcessiveLoad = LoadAvg > 2*DetectedCpus + 2
 
 # only accept new work for a while, for validated nodes and jobs with project name set
 # On GPU nodes, CPU only jobs are only allowed after all GPUs are in use
 START = (isUndefined(DaemonStartTime) || (time() - DaemonStartTime) < (MY.ACCEPT_JOBS_FOR_HOURS*60*60)) && \
-        (MY.OSG_NODE_VALIDATED == True) && \
+        (isUndefined(MY.OSG_NODE_VALIDATED) || MY.OSG_NODE_VALIDATED) && \
```
Comment (matyasselmeci) [inline 50-main.config:54]: This also runs jobs if OSG_NODE_VALIDATED is undefined - is that on purpose? When would OSG_NODE_VALIDATED be undefined?

Comment (rynge) [reply 50-main.config:54]: The main concern would be other VOs picking up the container and maybe using different validation scripts.

Comment (matyasselmeci) [reply 50-main.config:54]: Can you add a comment saying that, so I don't ask this question next time?

Comment (rynge) [reply 50-main.config:58]: Done

**Review by matyasselmeci — APPROVED** [review]
LGTM.

**Pull request merged**